sns/ut/isal_ub.c

```diff
@@ -0,0 +1,356 @@
+/*
+ * Copyright (c) 2011-2020 Seagate Technology LLC and/or its Affiliates
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * For any questions about this software or licensing,
+ * please email opensource@seagate.com or cortx-questions@seagate.com.
+ *
+ */
+
+
+#include <stdio.h>
+#include <stdlib.h>
+#include <string.h>
+#include <getopt.h>
+#include <isa-l.h>
+
+#include "lib/ub.h"
+#include "lib/misc.h"
+#include "lib/memory.h"
+#include "lib/buf.h"
+#include "lib/time.h"
+#include "lib/arith.h"
+
+#define UB_ITER 1
+#define MMAX    255
+#define KMAX    255
+
+m0_time_t seed = 0;
+
+static int ub_init(const char *opts M0_UNUSED)
+{
+    seed = m0_time_now();
+
+    return 0;
+}
+
+static void unit_spoil( struct m0_buf *failed,
+                        const uint32_t data_count )
+{
+    uint32_t err_id, i, j;
+    bool found;
+    uint8_t *err_list = (uint8_t *)failed->b_addr;
+
+    i = 0;
+    do {
+        err_id = m0_rnd(data_count, &seed);
+        found = false;
+
+        for (j = 0; j < i; j++) {
+            if (err_list[j] == err_id) {
+                found = true;
+                break;
+            }
+        }
+        if (!found) {
+            err_list[i++] = err_id;
+        }
+    } while ( i < failed->b_nob );
+}
+
+static void encode_data(uint32_t k, uint32_t p, uint32_t len,
+                        uint8_t *encode_matrix, uint8_t *g_tbls,
+                        uint8_t **fragments)
+{
+    // Initialize g_tbls from encode matrix
```

Comment:
Please use /* */ style of comments. No C++ comments.

Reply:
Done.